docs/src/user_docs/assembly/code_organization.md

```diff
@@ -3,35 +3,26 @@ A Miden assembly program is just a sequence of instructions each describing a sp
 
 In turn, Miden assembly instructions are just keywords which can be parameterized by zero or more parameters. The notation for specifying parameters is *keyword.param1.param2* - i.e., the parameters are separated by periods. For example, `push.123` instruction denotes a `push` operation which is parameterized by value `123`.
 
-Currently, Miden assembly provides two types of code organization blocks: *programs* and *procedures*. In the future, additional blocks such as modules and functions will be added.
-
-### Programs
-A *program* block is used to define an executable program. A program must start with a `begin` instruction and terminate with an `end` instruction. For example:
-```
-begin
-    <instructions>
-end
-```
-When Miden assembly code is executed, the execution starts at the first instruction following the `begin` instruction of the program. A program block is expected to be the last block in an assembly file and can be followed only by comment blocks.
+Miden assembly programs are organized into procedures. Procedures, in turn, can be grouped into modules.
 
 ### Procedures
-A *procedure* block is used to define a frequently-used sequence of instructions. A procedure must start with a `proc.<label>.<number of locals>` instruction and terminate with an `end` instruction. For example:
+A *procedure* can be used to encapsulate a frequently-used sequence of instructions which can later be invoked via a label. A procedure must start with a `proc.<label>.<number of locals>` instruction and terminate with an `end` instruction. For example:
 ```
 proc.foo.2
     <instructions>
 end
 ```
-A procedure label must start with a letter and can contain any combination of numbers, ASCII letters, and underscores (`_`).
+A procedure label must start with a letter and can contain any combination of numbers, ASCII letters, and underscores (`_`). The number of characters in the procedure label cannot exceed 100.
 
-The number of locals specifies the number of memory-based local words a procedure can access (via `load.local`, `store.local`, and other instructions). If a procedure doesn't need any memory-based locals, this parameter can be omitted or set to `0`. The number of locals per procedure is not limited, but the total number of locals available to all procedures at runtime must be smaller than $2^{32}$.
+The number of locals specifies the number of memory-based local words a procedure can access (via `loc_load`, `loc_store`, and [other instructions](./io_operations.md#random-access-memory)). If a procedure doesn't need any memory-based locals, this parameter can be omitted or set to `0`. A procedure can have at most $2^{16}$ locals, and the total number of locals available to all procedures at runtime is limited to $2^{30}$.
 
-To execute a procedure an `exec.<label>` instruction should be used. For example:
+To execute a procedure, `exec.<label>`, `call.<label>`, and `syscall.<label>` instructions can be used. For example:
```

Comment:
nit: this would be easier to read as "To execute a procedure, the `exec.<label>`, ..."